pkg/ingester/instance_test.go

```diff
@@ -0,0 +1,41 @@
+package ingester
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/grafana/loki/pkg/logproto"
+
+	"github.com/grafana/loki/pkg/util/validation"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLabelsCollisions(t *testing.T) {
+	o, err := validation.NewOverrides(validation.Limits{MaxStreamsPerUser: 1000})
+	require.NoError(t, err)
+
+	i := newInstance("test", 512, o)
+
+	tt := time.Now().Add(-5 * time.Minute)
+
+	// Notice how labels aren't sorted.
+	err = i.Push(context.Background(), &logproto.PushRequest{Streams: []*logproto.Stream{
```

Comment:
here 2 push in different goroutines, will also help catch race condition.

Reply:
Added another test for testing concurrent pushes.